### ISSUE TITLE
ESLint disable comments and disabled/non standard ESLint config rules

### ISSUE THREAD

**Issue by peterschriever** [hyva-themes/magento2-hyva-checkout#111]
Why is that while ESLint is being used we are also adjusting and disabling so many of its rulings?   

Snippet from the eslint config:
```javascript
  rules: {
    'arrow-body-style': [2, 'as-needed'],
    'class-methods-use-this': 0,
    'import/imports-first': 0,
    'import/newline-after-import': 0,
    'import/no-dynamic-require': 0,
    'import/no-extraneous-dependencies': 0,
    'import/no-named-as-default': 0,
    'import/no-unresolved': 2,
    'import/no-webpack-loader-syntax': 0,
    'import/prefer-default-export': 0,
    indent: [
      2,
      2,
      {
        SwitchCase: 1,
      },
    ],
    'jsx-a11y/aria-props': 2,
    'jsx-a11y/heading-has-content': 0,
    'jsx-a11y/label-has-associated-control': 2,
    'jsx-a11y/mouse-events-have-key-events': 2,
    'jsx-a11y/role-has-required-aria-props': 2,
    'jsx-a11y/role-supports-aria-props': 2,
    'max-len': 0,
    'newline-per-chained-call': 0,
    'no-confusing-arrow': 0,
    'no-console': 1,
    'no-use-before-define': 0,
    'prefer-template': 2,
    'react/jsx-closing-tag-location': 0,
    'react/forbid-prop-types': 0,
    'react/jsx-first-prop-new-line': [2, 'multiline'],
    'react/jsx-filename-extension': 0,
    'react/jsx-no-target-blank': 0,
    'react/require-default-props': 0,
    'react/require-extension': 0,
    'react/self-closing-comp': 0,
    'react/sort-comp': 0,
  },
```

And when searching in the code there are **35** `// eslint-disable` comments! 35! That doesnt indicate some exceptional cases, thats a problem which sets a poor example to any developer working with this codebase.

**Comment by progammer-rkt** [comment]
Okay. I thoroughly checked this. Most of the rule customization present here is either not needed or else already enforced by Airbnb coding standards. 

I cleared the eslint rules now and fixed all related issues in the codebase. This work is available in #134

**Comment by progammer-rkt** [comment]
The new `eslint` rules now reached `develop` branch and hence, closing this issue
